From 17420c9c64d68bcbd42dd76542e97396002bb372 Mon Sep 17 00:00:00 2001
From: Matt Ranostay <mranostay@gmail.com>
Date: Sun, 4 Aug 2013 22:25:08 +0000
Subject: [PATCH 09/40] bone: add protocape GPS

Add prototype GPS cape with UART and pps-gpio fragments

Signed-off-by: Matt Ranostay <mranostay@gmail.com>
---
 arch/arm/boot/dts/am335x-bone-common.dtsi | 20 ++++++++++-
 firmware/Makefile                         |  1 +
 firmware/capes/BB-BONE-GPS-00A0.dts       | 59 +++++++++++++++++++++++++++++++
 3 files changed, 79 insertions(+), 1 deletion(-)
 create mode 100644 firmware/capes/BB-BONE-GPS-00A0.dts

diff --git a/arch/arm/boot/dts/am335x-bone-common.dtsi b/arch/arm/boot/dts/am335x-bone-common.dtsi
index 1e86cd5..f144957 100644
--- a/arch/arm/boot/dts/am335x-bone-common.dtsi
+++ b/arch/arm/boot/dts/am335x-bone-common.dtsi
@@ -202,6 +202,15 @@
 				part-number = "BB-BONE-RS232-01";
 			};
 
+			/* GPS cape */
+			slot@13 {
+				compatible = "kernel-command-line", "runtime";
+				board-name = "BB-BONE-GPS";
+				version = "00A0";
+				manufacturer = "Adafruit";
+				part-number = "BB-BONE-GPS-01";
+			};
+
 			/* Beaglebone black has it soldered on */
 			 slot@100 {
 				ti,cape-override;
@@ -366,8 +375,17 @@
 				};
 			};
 
-			/* beaglebone black hdmi on board (No audio) */
 			cape@13 {
+				part-number = "BB-BONE-GPS-01";
+				version@00A0 {
+					version = "00A0";
+					dtbo = "BB-BONE-GPS-00A0.dtbo";
+				};
+			};
+
+
+			/* beaglebone black hdmi on board (No audio) */
+			cape@14 {
 				part-number = "BB-BONELT-HDMIN";
 				version@00A0 {
 					version = "00A0";
diff --git a/firmware/Makefile b/firmware/Makefile
index f4331ff..fa63014 100644
--- a/firmware/Makefile
+++ b/firmware/Makefile
@@ -156,6 +156,7 @@ fw-shipped-$(CONFIG_CAPE_BEAGLEBONE) += \
 	BB-BONE-eMMC1-01-00A0.dtbo \
 	BB-BONE-GPEVT-00A0.dtbo \
 	BB-BONE-RTC-00A0.dtbo \
+	BB-BONE-GPS-00A0.dtbo \
 	BB-BONE-RS232-00A0.dtbo \
 	BB-BONE-SERL-01-00A1.dtbo \
 	cape-bone-iio-00A0.dtbo \
diff --git a/firmware/capes/BB-BONE-GPS-00A0.dts b/firmware/capes/BB-BONE-GPS-00A0.dts
new file mode 100644
index 0000000..83103d6
--- /dev/null
+++ b/firmware/capes/BB-BONE-GPS-00A0.dts
@@ -0,0 +1,59 @@
+/*
+ * Copyright (C) 2013 Matt Ranostay <mranostay@gmail.com>
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2 as
+ * published by the Free Software Foundation.
+ */
+/dts-v1/;
+/plugin/;
+
+/ {
+	compatible = "ti,beaglebone", "ti,beaglebone-black";
+
+	/* identification */
+	part-number = "BB-BONE-GPS";
+	version = "00A0";
+
+	fragment@0 {
+		target = <&am33xx_pinmux>;
+		__overlay__ {
+			uart_pins: pinmux_uart_pins {
+				pinctrl-single,pins = <
+					0x150	0x21	/* spi0_sclk.uart2_rxd | MODE1 | PULL_UP */
+					0x154	0x01	/* spi0_d0.uart2_txd | MODE1 */
+				>;
+			};
+
+			pps_pins: pinmux_pps_pins {
+				pinctrl-single,pins = <
+					0x040	0x27	/* gpmc_a0.gpio1_16, INPUT | PULLDIS | MODE7 */
+				>;
+			};
+		};
+	};
+
+	fragment@1 {
+		target = <&uart3>;
+		__overlay__ {
+			status = "okay";
+			pinctrl-names = "default";
+			pinctrl-0 = <&uart_pins>;
+		};
+	};
+
+	fragment@2 {
+		target = <&ocp>;
+		__overlay__ {
+			pps {
+				compatible = "pps-gpio";
+				status = "okay";
+				pinctrl-names = "default";
+				pinctrl-0 = <&pps_pins>;
+
+				gpios = <&gpio2 16 0>;
+				assert-falling-edge;
+			};
+		};
+	};
+};
-- 
1.9.0

